### PR TITLE
Add comments to our interfaces and abstract classes

### PR DESCRIPTION
**Pull request by leofeyer**
This PR adds comments to our interfaces and abstract classes so people know how these will change in Contao 5. It is pretty much the same thing that Symfony did.

### CODE REVIEW

File: core-bundle/src/Command/AbstractLockedCommand.php [hunk 1]
```diff
@@ -70,7 +70,7 @@ final protected function execute(InputInterface $input, OutputInterface $output)
     /**
      * @return int
      */
-    abstract protected function executeLocked(InputInterface $input, OutputInterface $output);
+    abstract protected function executeLocked(InputInterface $input, OutputInterface $output)/*: int*/;
```
Comment (fritzmg) [inline core-bundle/src/Command/AbstractLockedCommand.php]: This class will be removed in Contao 5.0 anyway btw.

Comment (leofeyer) [reply core-bundle/src/Command/AbstractLockedCommand.php]: Reverted in 8a11b062f49367e50172992bca97f89b629824ad.

File: installation-bundle/src/Database/AbstractVersionUpdate.php [hunk 2]
```diff
@@ -37,15 +37,15 @@ public function __construct(Connection $connection)
     /**
      * @return bool
      */
-    public function hasMessage()
+    public function hasMessage()/*: bool*/
```
Comment (fritzmg) [inline installation-bundle/src/Database/AbstractVersionUpdate.php]: This class is also deprecated 🙃

**Comment by aschempp** [comment]
what's the benefit of making this effort? I mean no developer will ever see this except by looking at the code, and the docblock already gives the same information?

**Comment by fritzmg** [comment]
The docblock does not specify the enforced type though. Within Contao at least we allow parameter and return types to be different from the docblock.

**Comment by leofeyer** [comment]
> what's the benefit of making this effort?

Why did Symfony do it? And why did we do it in the `PictureFactoryInterface`?

https://github.com/contao/contao/blob/bfff2382777eb19e9a16697f3b8db9b4d37b60e1/core-bundle/src/Image/PictureFactoryInterface.php#L40

**Review by ausi — APPROVED** [review]

**Comment by aschempp** [comment]
should we also set `#[\ReturnTypeWillChange]` ?

**Pull request merged**